File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res = 0
        j = 0
        key_count = defaultdict(int)
        for i in range(len(s)):
            key_count[s[i]] += 1
            while key_count[s[i]] > 1:
                print(j)
                key_count[s[j]] -= 1
                j += 1
            res = max(res, i - j + 1)
        return res

File: Data Structures & Algorithms/count-subsequences/submission-1.cpp
class Solution {
public:
    int numDistinct(string s, string t) {
        vector<vector<int>> res(s.size() + 1, vector<int>(t.size() + 1, 0));
        for(int i = 0; i < res.size(); ++i) {
            res[i][res[0].size() - 1] = 1;
        }
        for(int i = res.size() - 2; i > -1; --i) {
            for(int j = res[0].size() - 2; j > -1; --j) {
                if(s[i] == t[j]) {
                    res[i][j] = res[i + 1][j + 1] + res[i + 1][j];
                } else {
                    res[i][j] = res[i + 1][j];
                }
            }
        }
        return res[0][0];
    }
};
